server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -16,183 +16,185 @@ def get(self, section: str, option: str):
         res = ptb.process_tb()
 
         assert res == expected_result
 
-    def test_process_tb_zero(self, monkeypatch):
+    def test_process_tb_zero(self, monkeypatch, test_logger):
         """verify processing if there are no TBs without any failure"""
         tb_path = fixture_path / "tarball"
 
-        def mock_glob(*args, **kwargs):
+        def mock_glob(self, pattern):
             return []
 
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
+
+        def mock_results_push(self, ctrl, tb, token):
+            assert False, "Unexpected call to mocked result_push function"
 
         expected_result = Results(nstatus="", ntotal=0, ntbs=0)
         monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
 
-    def test_multiple_process_tb(self, monkeypatch, caplog):
+    def test_multiple_process_tb(self, monkeypatch, caplog, test_logger):
         """verify tar balls processing at the time of Failure as well as success"""
-        tb_path = fixture_path
-
-        def mock_remove(*args):
-            pass
+        tb_path = fixture_path / "mix-tarball"
+        good_tb = tb_path / tarball
+        bad_tb = tb_path / bad_tarball
 
-        def mock_glob(*args, **kwargs):
-            tb_path1 = fixture_path / "tarball-002" / "log.tar.xz.md5"
-            tb_path2 = fixture_path / "bad_tarball-002" / "bad_log.tar.xz.md5"
-            return [tb_path1, tb_path2]
+        def mock_remove(path, *, dir_fd=None):
+            if path == good_tb or path == bad_tb:
+                pass
+            assert False, "Unexpected tar ball in mocked os.remove()"
 
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
-                return tb_path / "tarball"
+                return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         def mock_results_push(self, ctrl, tb, token):
-            if not tb.exists():
+            bad_tb = self.receive_dir / bad_tarball
+            if bad_tb == tb:
                 raise FileNotFoundError(f"No such file or directory: '{tb.name}'")
+            return
```

Comment:
Since `mock_results_push()` doesn't return a value, you can omit the `return` statement from the end.

(Also, type hints in the function signature would be good.  And, I don't think it's worth creating the `bad_tb` local variable.)

Reply:
then how should I compare it? Should I do this directly?

Reply:
The main reason for using `self.receive_dir` was to use tb_path with appended `-002` suffix.

Reply:
Understood.  However, depending on values from the CUT in order to validate the behavior of the CUT is a little bit like letting the inmates run the asylum.  🙂

The test code should be able to predict what the value of `self.receive_dir` will be, in which case, it should be able to predict what the final values of `good_tb` and `bad_tb` will be, which allows it to confirm that the CUT behaves as expected.